### ISSUE TITLE
Preserve more state across restarts

### ISSUE THREAD

**Issue by fabiospampinato** [fabiospampinato/notable#339]
### Feature description

Some more state should be preserved across restarts:

- [ ] If edit mode is active
- [ ] If focus mode is active
- [ ] If split-view mode is active
- [ ] What the active tag is (#326)
- [ ] What the active note is
- [ ] The widths of the columns (#206)
- [ ] If line wrapping is enabled

Some of these should be data-directory-specific, but maybe initially I'll implement them assuming that there's only 1 data directory being used and then I'll generalize it later.

### Feature motivation

Making the same changes over and over again becomes annoying pretty quickly.

**Comment by Trenar** [comment]
I think preserving the "collapsed" or "expanded" state of tags in the left panel is helpful, too. I love your idea of indefinite nestable tags, but most of the time I collapse them at the first or second level at restart of the app

**Comment by RomanKornev** [comment]
Also
- [ ] Active note scroll position
- [ ] Active note cursor position
- [ ] Window position
- [ ] Active search query
- [ ] If full screen mode is active

**Comment by godzeus** [comment]
As @Trenar commented
> I think preserving the "collapsed" or "expanded" state of tags in the left panel is helpful, too. I love your idea of indefinite nestable tags, but most of the time I collapse them at the first or second level at restart of the app

I will like the idea to remember the collapsed or expanded state for tags/notebooks. By the way I founded the app very useful. Thanks!

**Comment by godzeus** [comment]
As @Trenar commented
> I think preserving the "collapsed" or "expanded" state of tags in the left panel is helpful, too. I love your idea of indefinite nestable tags, but most of the time I collapse them at the first or second level at restart of the app

I will like the idea to remember the collapsed or expanded state for tags/notebooks. By the way I found the app very useful. Thanks!

**Comment by fabiospampinato** [comment]
I've added a few more things to restore to the main post, thanks all for the suggestions! I'm not sure about some of them:

> Active note scroll position
> Active note cursor position
> Active search query

Maybe we should restore those, but right now I think it would be better if the app started in a bit more pristine state.

Also:

>  Window position

@RomanKornev Isn't this already the case? 🤔

**Comment by RomanKornev** [comment]
> @RomanKornev Isn't this already the case? 🤔

@fabiospampinato Just checked. Seems to be fixed on master.

**Comment by amelandri** [comment]
Also sidebar width and note list width should be preserved across restarts

**Comment by fabiospampinato** [comment]
> Also sidebar width and note list width should be preserved across restarts

That's already tracked in the "The widths of the column" point.

**Comment by Armagedoom** [comment]
IMO the 'Sorting by' (title, date created, date modified)  should also be persistent among tags and restarts.
Reason being, there are certain tags in which you will always want to see the last edited one, others you will want always to see them sorted alphabetically, etc.
Thanks!

**Comment by fabiospampinato** [comment]
@Armagedoom That's interesting 🤔 I'll think about it.

**Comment by zawnk** [comment]
I agree! Especially with `What the active note is`. I'm trying to integrate Notable in my workflow at work and I think it always opens the note that's highest in the list, which is in my case a note with sensitive information. This is quickly becoming annoying, when I can't open my notes in a meeting or something when it opens the note with sensitive information. So Notable just opening the last note that was open on restart would be great! 

Any update/progress or idea where it's roughly on the roadmap to know when and what to expect?

Thanks :)

**Comment by fabiospampinato** [comment]
> Any update/progress or idea where it's roughly on the roadmap to know when and what to expect?

I'm currently in the process of writing our core settings library, which will be used for storing these sorts of data as well, it will land on v1.8 so probably this will be implemented in v1.9 if I had to guess.

**Comment by fabiospampinato** [comment]
Quick update: there's been significant progress on this, now that proper decent settings are implemented preserving some state across restarts, even at the data-directory level, is easy, if there's any state that's not mentioned already [here](https://github.com/notable/notable/issues/339#issue-402509683) that you think the app should preserve this is the right time to mention it.

**Comment by Armagedoom** [comment]
- [ ] TLDR: Keep scroll position across restarts and edit/view mode of the notes as @RomanKornev  suggested.

Long version:
After using images with Notable, I found out that having a persistent active note scroll position is quite important for when having images in a node, specially across edit / view mode.
Since the images by default are very big and fill in the page, either they are left to the bottom of the document (which shouldn't be forced just because of display issues) or you end up having to scroll down a lot every time you open (or switch between edit / display mode) for that note, which is very inconvenient and forced me to edit several notes just to put the images on the bottom.
IMO the behavior should be such as not to force the user to arrange the documents arround it but to ease him to be able to use them as he pleases.
So, if we were to keep the scroll position for a note (even after restarts) notes with big amount of images wouldnt be a hell to traverse and we wouldnt need to put all the images at the bottom of the document.

**Comment by fabiospampinato** [comment]
@Armagedoom We are tracking that in the following task:

> - [ ] Previewing/editing state for all notes

That's a bit more complicated to do than the simple bits of states the app is preserving right now, mainly because we are talking a list of potentially complicated objects rather than a simple boolean, and also those states internally are uniquely identified by a note's internal unique id, which isn't exposed at the moment, so notes need to be identified differently when saving their state. It's not really complicated to do, if this won't get implemented for v1.9 already it'll surely come in v1.10 I think.

**Comment by Armagedoom** [comment]
Oh I didnt notice. This is not a truly blocker issue but its good to know it will eventually arrive. Thanks!
